src/core/MIGRATION.md

```diff
@@ -1115,6 +1116,164 @@ class MyPlugin {
   }
 ```
 
+### Handle HTTP request with New Platform HTTP Service
+Kibana HTTP Service provides own abstraction for work with HTTP stack.
+Plugins don't have direct access to `hapi` server and its primitives anymore. Moreover,
+you shouldn't rely on the fact that HTTP Service uses one or another library under the hood, there are high chances that we can switch to another stack at all. If HTTP Service
+lacks needed functionality we happy to discuss and support your needs.
+
+To handle an incoming request in your plugin you should:
+- create Router instance. Use `plugin id` as a prefix path segment for your routes.
```

Comment:
```suggestion
- create a `Router` instance. Use `plugin-id` as a prefix path segment for your routes.
```

**optional:** we could provide a getRouter function to plugins that automatically uses their plugin id as the prefix?

Reply:
yes, we can. `Router` is a class exported from `src/core/server`. because `Router` belongs to the `HTTP` domain, we can make it a part of `HTTP service` contract and expose a factory function which already prefixed with `plugin-id`.